OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseUnitWebViewFragment.java

```diff
@@ -96,6 +119,103 @@ public void onPageLoadProgressChanged(WebView webView, int progress) {
+        // TODO UPGRADE_TO_GRADED & UPGRADE_TO_RESET will be enable once we are allowed to do payment through mobile
+        switch (courseBannerInfo.getDatesBannerInfo().getCourseBannerType()) {
+            case UPGRADE_TO_GRADED:
+                textView.setText(getText(R.string.course_dates_banner_upgrade_to_graded));
+                break;
+            case UPGRADE_TO_RESET:
+                textView.setText(getText(R.string.course_dates_banner_upgrade_to_reset));
+                break;
+            case RESET_DATES:
+                textView.setText(getText(R.string.course_dates_banner_reset_date));
+                buttonText = getContextOrThrow().getString(R.string.course_dates_banner_reset_date_button);
+                break;
+            case INFO_BANNER:
+                textView.setText(getText(R.string.course_dates_info_banner));
+                break;
+            case BLANK:
+                infoBanner.setVisibility(View.GONE);
+                break;
+        }
+
+        if (!TextUtils.isEmpty(textView.getText())) {
+            if (!TextUtils.isEmpty(buttonText)) {
+                button.setText(buttonText);
+                button.setVisibility(View.VISIBLE);
+                button.setOnClickListener(v -> {
+                    courseDateViewModel.resetCourseDatesBanner(unit.getCourseId());
+                });
+            }
+            infoBanner.setVisibility(View.VISIBLE);
+        }
+    }
+
+    private void initObserver() {
+        courseDateViewModel.getBannerInfo().observe(this, this::initInfoBanner);
+
+        courseDateViewModel.getShowLoader().observe(this, flag ->
+                preloadingListener.setLoadingState(flag ?
+                        PreLoadingListener.State.MAIN_UNIT_LOADING :
+                        PreLoadingListener.State.MAIN_UNIT_LOADED));
+
+        courseDateViewModel.getResetCourseDates().observe(this, resetCourseDates -> {
+            if (resetCourseDates != null) {
+                showShiftDateSnackBar(true);
+                canFetchBannerInfo = false;
+                authWebView.loadUrl(true, unit.getBlockUrl());
+            }
+        });
+
+        courseDateViewModel.getErrorMessage().observe(this, errorMessage -> {
+            if (errorMessage != null) {
+                if (errorMessage.getThrowable() instanceof AuthException || errorMessage.getThrowable() instanceof HttpStatusException &&
+                        ((HttpStatusException) errorMessage.getThrowable()).getStatusCode() == HttpStatus.UNAUTHORIZED) {
+                    environment.getRouter().forceLogout(getContextOrThrow(),
+                            environment.getAnalyticsRegistry(),
+                            environment.getNotificationDelegate());
+                } else {
+                    switch (errorMessage.getErrorCode()) {
+                        case ErrorMessage.BANNER_INFO_CODE:
+                            initInfoBanner(null);
```

Comment:
We have to create the util for  `initInfoBanner` & `showShiftDateSnackBar` using at muitple classes.

Reply:
I have implemented the initInfoBanner in utils class but for snackbar we have to pass too many arguments so I avoid it